### ISSUE TITLE
unmet peer dependency when I have graphql 0.10.3

### ISSUE THREAD

**Issue by thunder7553** [dotansimha/graphql-code-generator#121]
I get an unmet peer dependency, but I have graphql 0.10.3 installed (node 8+, npm 5+)

``` sh
graphql-code-generator@0.5.4 requires a peer of graphql@^0.10 | ^0.9 | ^0.8 but none was installed.
```

``` sh
>npm --version
5.0.3
>node --version
v8.1.3
>
```

I think the problem is that the versions in package.json do not include the trailing '.0'. The other problem - please correct me if I'm wrong - the ^ operator should match any change of minor and patch, according to this: https://nodesource.com/blog/semver-tilde-and-caret/. So ^0.8.0 would match anything lower than 1.0.0. Changes in the patch number only would require the ~ operator: ~0.10.0 | ~0.9.0 | ~0.8.0 would work for any 0.8, 0.9 and 0.10, but not for 0.11.

**Comment by thunder7553** [comment]
```yaml
    "graphql": ">=0.8.0 < 1.0.0"
```

seems to be working

**Comment by dotansimha** [comment]
Fixed in 0.5.5 :) Thanks @thunder7553
